dbt/include/trino/macros/materializations/materialized_view.sql

```diff
@@ -0,0 +1,80 @@
+{% materialization trino_materialized_view, adapter="trino" %} 
+{%- set target_relation = this %}
+{%- set existing_relation = load_relation(this) -%}
+{%- set config_grace_period = config.require('grace_period') %}
+{%- set config_max_import_duration = config.require('max_import_duration') %}
+{%- set config_drop_any_existing = config.require('drop_any_existing') %}
+{%- set config_cron = config.get('cron') %}
+{%- set config_refresh_interval = config.get('refresh_interval') %}
+
+
+{{ run_hooks(pre_hooks) }}
+
+{% if existing_relation is none %}
+
+    {{ log("No existing view found, creating materialized view...", info=true) }}
+    {%- set build_sql = build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) %}
+
+{% else %}
+    {%- if existing_relation.is_view %}
+        {%-set existing_type = 'view'%}
+    {%- elif existing_relation.is_table %}
+        {%-set existing_type = 'table'%}
+    {%- elif existing_relation.is_cte %}
+        {%-set existing_type = 'CTE'%}
+    {% endif %}
+
+    {%- if existing_type is not none and config_drop_any_existing == 'False' %}
+        {{ log("Found a " ~ existing_type ~ " with same name.", info=true) }}
+        {%- set build_sql = build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) %}
+
+    {% elif existing_type is not none and config_drop_any_existing == 'True' and existing_type != 'table'%} -- theres an issue with starburst thinking that the materialised view is a table in the information schema, hence the logic on this line
+        {%- set drop_existing_sql = "DROP " ~ existing_type ~ " IF EXISTS " ~ existing_relation %}
```

Comment:
Makes me wonder if this should not be more generally applied. For example a MV can be replaced by a table.

Reply:
This line is something i need to look into further at some point, theres an issue where it thinks an existing mat view is a table coming from starburst, hence its excluding table on this line, so causes an error due to recieving the wrong type, the error flow looks like this

- Finds an existing item
-E xisting item is a materialised view, however schema tables in starburst have marked it as a table incorrectly
- Decides to run this code to drop a table
- Error occurs as it runs DROP TABLE against a MATERIALIZED VIEW

So this currently ignores table types and will display an error in console informing user a table already exists and they need to manually handle it (drop, move etc..), otherwise itll run fine if its a materialized view incorrectly typed, due to the create or replace part of the SQL statement further on

Reply:
These are all good cases that should be integration tested.

Reply:
See following PR https://github.com/trinodb/trino/pull/15350 where support in system table tables will be improved with the type.

Reply:
That mentioned PR was reverted here: https://github.com/trinodb/trino/pull/16143